noggin/representation/user.py

```diff
@@ -67,3 +67,24 @@ def rhbz_mail(self):
     @property
     def last_password_change(self):
         return self._attr('krblastpwdchange')
+
+    def get_properties(self):
+        """
+        Returns a dict of all the properties defined above and their values
+        """
+        return {
+            key: getattr(self, key)
+            for key, value in self.__class__.__dict__.items()
+            if isinstance(value, property)
+        }
+
+    def diff_against(self, other):
+        """
+        Compares two instances of the User class, and returns the properties
+        with values that are different
+        """
+        return [
+            key
+            for key, value in self.get_properties().items()
```

Comment:
Why not use the `.raw` attribute? And do something like `for key, value in self.raw.items()`

Reply:
.raw contains the ipa names, i.e. "uid" rather than "username",that is why i added get_properties()

Reply:
Ah, I see. It still feels a bit "unnatural" to do it this way, I'll try another way, maybe using `__iter__`.